src/app.ts

```diff
@@ -13,14 +13,40 @@ interface TemperatureSummary {
   average: number
 }
 
+let reading: TemperatureReading[];
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  reading = readings;
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
-}
+  let totalTemp: number = 0;
+  let counter: number = 0;
+  let founded: boolean = false;
+  const obj: TemperatureSummary = {
+    first: 0,
+    last: 0,
+    high: 0,
+    low: 0,
+    average: 0
+  };
+
+  reading.forEach(r => {
```

Comment:
You're iterating the whole array while you check the conditions (i.e. date and city).

But, what if you use [array.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)?